src/components/TrafficList/TrafficListComponent.tsx

```diff
@@ -0,0 +1,353 @@
+      return getTrafficHealth(item, item.direction);
+    }
+
+    return { value: 0, status: NA };
+  };
+
+  private getTraffic = (traffic: ProtocolTraffic): { trafficRate; trafficPercent } => {
+    let rps = '0';
+    let percentError = '0';
+    let unit = 'rps';
+    if (hasProtocolTraffic(traffic)) {
+      switch (traffic.protocol) {
+        case 'http':
+          rps = traffic.rates.http;
+          percentError = traffic.rates.httpPercentErr || '0';
+          break;
+        case 'grpc':
+          rps = traffic.rates.grpc;
+          percentError = traffic.rates.grpcPercentErr || '0';
+          break;
+        case 'tcp':
+          rps = traffic.rates.tcp;
+          break;
+      }
+    }
+
+    return {
+      trafficRate: `${Number(rps).toFixed(2)}${unit}`,
+      trafficPercent: `${(100 - Number(percentError)).toFixed(1)}%`
+    };
+  };
+
+  // Helper used to build the table content.
+  rows = (direction: TrafficDirection): IRow[] => {
+    return this.state.listItems
+      .filter(i => i.direction === direction)
+      .map((item, i) => {
+        const name = item.node.name;
+        const links = this.getLinks(item);
+        return {
+          cells: [
+            <>
+              <Tooltip
+                key={`tt_status_${i}`}
+                position={TooltipPosition.top}
+                content={<>Traffic Status: {item.healthStatus.status.name}</>}
+              >
+                {createIcon(item.healthStatus.status, 'sm')}
+              </Tooltip>
+            </>,
+            <>
+              <Tooltip
+                key={`tt_badge_${i}`}
+                position={TooltipPosition.top}
+                content={<>{this.nodeTypeToType(item.node.type)}</>}
+              >
+                <Badge className={'virtualitem_badge_definition'}>{item.icon}</Badge>
+              </Tooltip>
+              {!!links.detail ? (
+                <Link key={`link_d_${item.icon}_${name}`} to={links.detail} className={'virtualitem_definition_link'}>
+                  {name}
+                </Link>
+              ) : (
+                name
+              )}
+            </>,
+            <>{item.trafficRate}</>,
+            <>{item.trafficPercent}</>,
+            <>{item.protocol}</>,
+            <>
+              {!!links.metrics && (
+                <Link key={`link_m_${item.icon}_${name}`} to={links.metrics} className={'virtualitem_definition_link'}>
+                  View metrics
+                </Link>
+              )}
+            </>
+          ]
+        };
+      });
+  };
+
+  private getLinks = (item: TrafficListItem) => {
+    if (item.node.isInaccessible) {
+      return { detail: '', metrics: '' };
+    }
+
+    const detail = `/namespaces/${item.node.namespace}/${this.nodeTypeToType(item.node.type, true)}/${item.node.name}`;
+
+    const metricsDirection = item.direction === 'inbound' ? 'in_metrics' : 'out_metrics';
+    let metrics = `${history.location.pathname}?tab=${metricsDirection}`;
```

Comment:
Looks like the behavior of the `View metrics` link is broken.
It switches to the metrics page, but it no longer pre-selects the right filters.

I think this has been broken for a while, as I see labels have changed (e.g.: `destination_app` is now `destination_canonical_service`). For this PR, I think it's OK to leave as is, because there is no regression. But, probably, we should open a bug.

Reply:
I'll just fix it now.

Reply:
OK, the 'View metrics' link for apps seems to be working.